src/lib/protocols/egd.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * egd.c
+ *
+ * Ethernet Global Data
+ * 
+ * Copyright (C) 2024 - ntop.org
+ * Copyright (C) 2024 - V.G <v.gavrilov@securitycode.ru>
+ *
+ * This file is part of nDPI, an open source deep packet inspection
+ * library based on the OpenDPI and PACE technology by ipoque GmbH
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_EGD
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+static void ndpi_search_egd(struct ndpi_detection_module_struct *ndpi_struct, struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+
+  NDPI_LOG_DBG(ndpi_struct, "search Ethernet Global Data\n");
+
+  if (packet->payload_packet_len > 32 && ntohs(get_u_int16_t(packet->payload, 0)) == 0x0D01)
+  {
+    if (get_u_int32_t(packet->payload, 4) == packet->iph->saddr && /* ProducerID */
+        ntohl(get_u_int32_t(packet->payload, 24)) < 31 && /* Status */
+        get_u_int32_t(packet->payload, 28) == 0) /* Reserved */
+    {
+      NDPI_LOG_INFO(ndpi_struct, "found Ethernet Global Data\n");
+      ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_EGD, NDPI_PROTOCOL_UNKNOWN, NDPI_CONFIDENCE_DPI);
+      return;
+    }
+  }
+
+  NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+}
+
+void init_egd_dissector(struct ndpi_detection_module_struct *ndpi_struct,u_int32_t *id)
+{
+  ndpi_set_bitmask_protocol_detection("EthernetGlobalData", ndpi_struct, *id,
+                                      NDPI_PROTOCOL_EGD,
+                                      ndpi_search_egd,
+                                      NDPI_SELECTION_BITMASK_PROTOCOL_UDP_WITH_PAYLOAD,
```

Comment:
Could you add an explicit comment about the fact that this protocol seems to be ipv4 only, please?